src/Microsoft.PowerShell.Commands.Utility/commands/utility/Sort-Object.cs

```diff
@@ -157,20 +157,19 @@ private int Heapify(List<OrderByPropertyEntry> dataToSort, OrderByPropertyCompar
                 }
 
                 // If we're doing a unique sort and the entry is not unique, discard the duplicate entry
-                if (Unique && !uniqueSet.Add(dataToSort[dataIndex]))
+                if (Unique && !uniqueSet.Add(dataToSort[dataIndex + discardedDuplicates]))
```

Comment:
It seems we could have `discardedDuplicates` as absolute value and exclude the additions in loop.

Reply:
I tried to keep with the previous code's practice with it calculated inline.  I don't know the impacts of one vs the other so I left that part alone.